nested-types.ipynb

```diff
@@ -0,0 +1,915 @@
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[<pyarrow.Buffer address=0x4bdc4021040 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f80 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020e80 size=24 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020ec0 size=11 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f00 size=1 is_cpu=True is_mutable=True>,\n",
+       " <pyarrow.Buffer address=0x4bdc4020f40 size=5 is_cpu=True is_mutable=True>]"
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_4.buffers()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<nanoarrow.c_lib.CArrayView>\n",
+       "- storage_type: 'struct'\n",
+       "- length: 5\n",
+       "- offset: 0\n",
+       "- null_count: 1\n",
+       "- buffers[1]:\n",
+       "  - validity <bool[1 b] 11011000>\n",
+       "- dictionary: NULL\n",
+       "- children[2]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'string'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[3]:\n",
+       "      - validity <bool[1 b] 10111000>\n",
+       "      - data_offset <int32[24 b] 0 3 3 3 7 11>\n",
+       "      - data <string[11 b] b'joemarkjane'>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'int8'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[2]:\n",
+       "      - validity <bool[1 b] 11110000>\n",
+       "      - data <int8[5 b] 1 2 0 4 0>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(na.c_array(column_4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Map\n",
+    "\n",
+    "Map type represents nested data where each value is a variable number of key-item pairs. Its physical representation is the same as a list of `{key, item}` structs.\n",
+    "\n",
+    "The difference between a struct and a map type is that a struct holds the key in the schema therefore needs to be a string and the items are all equal type and stored in values buffer per each field. There can be multiple keys and therefore multiple values buffers. The map comparing to the struct has one child array holding all different keys that need to be all same type but not necessarily string and second child array holding all the values which also need to be the same type.\n",
```

Comment:
> [about the struct] ... and the items are all equal type ...

That is not fully correct (or worded a bit ambiguous). The values for a single key are indeed all of the same type (but that is true for map as well, so not a distinguishing characteristic), but the different keys can have different value types (while for the map, all values of all keys need to have the same type, as you later in the paragraph)

Reply:
Yeah, didn't word correctly and reading it again I see it reads wrongly. Will update!